### DISCUSSION TITLE
Is this compatible with BRP069B45?

### DISCUSSION THREAD

**Discussion by haissio** [revk/xESP32-Faikout#53] [General]
Just tried to connect a ESP32 flashed with this into the AC unit (FTXF35D5V1B) S21 port, but it seems not to be working.
This AC unit seems to require BRP069B45 wifi module.
The ESP32 gets power from pin 1 in the S21 socket (+5V), and I have tried to connect GPIO27 to the TX, and GPIO26 to RX without success.

**Comment by revk** [comment]
The defaults in the code (make menuconfig) are Tx (from module to a/c) on 26 and Rx (from a/c to module) on -27 (minus meaning inverted), maybe check you have them the right way around, and check not inverted, i.e. -27 not 27.

**Reply by haissio** [reply]
Yes, I have tried swapping the pins, and even recompiling the binaries with 27 instead of -27.
I'm not fully getting this thing about the inverted signal. Is the A/C unit supposed to be sending data with logic levels inverted?
Keep in mind that I'm cabling directly to the ESP32, I have no other circuits like your adaptor. If I get it right, in your board you are inverting the levels with a small MOSFET, and that's why you have the pin also inverted by default?

**Reply by revk** [reply]
Yes, the PCB design as a FET on Rx for protection (signal is 5V not 3.3V) and that inverts.

If not using that board, don't set the - for invert.

Another test would be to loop back the 26 and 27 pins and see if it logs a "lookback" event.

**Reply by haissio** [reply]
Really nice feature! Well, I moved the ESP32 to the computer and when crossing ports 26 and 27 it did not log anything different. It keeps trying to switch from UART to UART S21.
Thanks to this I noticed only a part of the firmware was uploaded, so I recompiled and uploaded all the bin files (bootloader, partition-table,...). Right after this it started logging the loopback, but it keeps saying offline when I move it to the A/C unit.
Any ideas on what else to check?

Thanks in advance.

**Reply by revk** [reply]
There is a debug setting you can enable via MQTT that logs what if actually sees. I think there is a dump setting that logs even more detail - that may give some clues. Beyond that I'd be looking at an oscilloscope.

**Reply by haissio** [reply]
I have managed to make it work. This is amazing!
I've found that tension provided/accepted in different units has enough variation to work on one unit while not on another, so I've finally used a 5v regulator for the power and a 3.3v<->5v bidirectional converter for the data.
Also, I had to disable the auto-upgrade feature on the config because it was downloading your binary updates after a few hours of using it.

Thanks for such a great project and all your work on it, as well as for your help! It's fantastic to be able to control the A/C units without dependency of the manufacturer servers.

Since I'm doing my own circuit and I will not buy your board, is there any other way to send you appreciation for the project? Ko-fi or something similar?

**Reply by revk** [reply]
Wow, I am surprised. What module are you using?

**Reply by haissio** [reply]
Clone of Wemos mini-D1-ESP32 from AzDelivery. Like the ones in this article, but the ESP32 module is labeled as AzDelivery also:
https://newscrewdriver.com/2022/09/03/wemos-d1-mini-esp32-derivative/

I was trying to power directly from the 5v pin in the S21 socket, but I noticed that the same module that was working in one A/C unit, was not even booting in the other one, so I tried using a DC-5v converter with the ~12v pin, and that worked flawless.

Again thanks for such a great project!
